src/main/learnocaml_main.ml

```diff
@@ -266,6 +266,15 @@ let process_html_file orig_file dest_file base_url =
          | `Start_element ((e, "script"), attrs) -> transform_tag e "script" attrs "src"
          | `Start_element ((e, "img"), attrs) -> transform_tag e "img" attrs "src"
          | `Start_element ((e, "a"), attrs) -> transform_tag e "a" attrs "href"
+         | `Start_element ((e,"div"),attrs) -> if noSecret
+              then (let make_secret_not_visible attrs =
+                      match attrs with
+                      |[] -> []
+                      |((a,b),c)::tail -> if b="id" && c= "secret-section"
+                        then ((a,b),c)::(("","style"),"display:none")::tail
+                        else attrs in
+                    `Start_element ((e,"div"),make_secret_not_visible attrs) )
+              else `Start_element ((e,"div"),attrs)
```

Comment:
I tested your PR successfully with a secret (or none) in `demo-repository/server_config.json`: the code behaves well at `learn-ocaml build` time (and the static deployment is still OK :).

In order to make the code more concise and more readable, you could use a standard higher-order function such as `List.mem` (i.e., member).

Also you may be interested in using a [guard](https://ocaml.org/releases/4.07/htmlman/expr.html#sec135). FYI, `match l with S e -> if no_secret then e' else S e | t -> t`
can also be written `match l with S e when no_secret -> e' | t -> t`.

So you could use this more compact version of your code:

```suggestion
         | `Start_element ((e, "div"), attrs)
              when no_secret && List.mem (("", "id"), "secret-section") attrs ->
            `Start_element ((e, "div"), (("", "style"), "display:none") :: attrs)
```